uni/ulsp/lsif.icn

```diff
@@ -0,0 +1,392 @@
+   #  vertex to a resultSet vertex.
+   #  <i>This is intended for internal use only!</i>
+   method createReferenceResultVertex(outID, property)
+      local lsif_table
+
+      lsif_table := table()
+      lsif_table["id"] := nextID
+      referenceID := nextID
+      lsif_table["type"] := "vertex"
+      lsif_table["label"] := "referenceResult"
+      write(outFile, tojson(lsif_table))
+      nextID := getNextID()
+      lsif_table := table()
+      lsif_table["id"] := nextID
+      lsif_table["type"] := "edge"
+      lsif_table["label"] := "textDocument/references"
+      lsif_table["outV"] := outID
+      lsif_table["inV"] := nextID-1
+      write(outFile, tojson(lsif_table))
+      nextID := getNextID()
+      createItemEdge(nextID-2, rangeID, documentID, property)
+   end
+
+   #  Creates an item edge that links a given range vertex to a definitionResult.
+   #  Params: outID - The ID for the definitionResult or referenceResult associated with this edge.
+   #          ItemRangeID - The ID of the range that this Item edge belongs to
+   #          docID - The ID for the document that this Item edge belongs to
+   #          property - Whether this edge belongs to definitions or references
+   #  <i>This is intended for internal use only!</i>
+   method createItemEdge(outID, ItemRangeID, docID, property)
+      local lsif_table, ranges
+
+      ranges := []
+      put(ranges, ItemRangeID)
+      lsif_table := table()
+      lsif_table["id"] := nextID
+      lsif_table["type"] := "edge"
+      lsif_table["label"] := "item"
+      lsif_table["outV"] := outID
+      lsif_table["inVs"] := ranges
+      lsif_table["shard"] := docID
+      if \property then lsif_table["property"] := property
+      write(outFile, tojson(lsif_table))
+      nextID := getNextID()
+   end
+
+   #  Creates an contains edge that links all of the ranges in a document to that document.
+   #  or all the documents to a project.
+   #  Params: outID - The ID that this edge is pointing out from, either the document or project ID.
+   #          contains_table - The table with ID's of all ranges in a document or documents in a project.
+   #  <i>This is intended for internal use only!</i>
+   method createContainsEdge(outID, contains_table)
+      local lsif_table
+
+      lsif_table := table()
+      lsif_table["id"] := nextID
+      lsif_table["type"] := "edge"
+      lsif_table["label"] := "contains"
+      lsif_table["outV"] := outID
+      lsif_table["inVs"] := contains_table
+      write(outFile, tojson(lsif_table))
+      nextID := getNextID()
+   end
+
+   #  Initially set this class' database to db and nextID to 1. This will also clear out the
+   #  output.lsif file and open it in append mode to be worked on by the Indexer.
+   initially(db)
+      self.db := db
+      self.nextID := 1
+      self.outFile := open(db.unicon_dir || "uni/ulsp/unicon.lsif", "w") | stop("Unable to open file")
```

Comment:
Do we really want to `stop()` here?  Does that mean that lsif can't be used without the lsif file ?